.github/workflows/ci.yml

```
@@ -397,6 +454,67 @@ jobs:
       - name: Build
         run: cmake --build build --config Release --verbose
 
+      - name: Zip build folder
+        run: zip -r build.zip ./build
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: Linux-build
```

You are using twice the same artifact name
```suggestion
          name: macOS-build
```

---

Sure.

---

I mean: that's the reason the CI was failing. Since it was both the same name, i.e. id, the upload artifact was trying to write concurrently the same space.